### PR TITLE
watch hosts and classification rules

### PR DESCRIPTION
**Pull request by dhellmann**
If we only watch the HardwareClassification resources, then we won't update hosts when they change. If we only watch the hosts, then we won't apply new classification rules. So, watch both.

This is still WIP because while it does trigger a host reconcile for each update, it does not apply the rules to figure out which labels to put on the host.

### CODE REVIEW

**Comment by dhellmann** [comment]
/cc @digambar15 @bfournie 

I've mentioned to each of you the issue I saw with the reconcile behavior. This branch shows how to set up a new controller to watch both types of resources. Eventually the logic for managing the labels should be used in the new controller, and the old one could probably be removed.

**Comment by dhellmann** [comment]
@digambar15 I think this is ready for review. The unit test coverage in the current draft includes the low-level functions but not the controllers. There weren't really tests for the controller Reconcile() function before, but I can add some to this PR if you want me to.

**Comment by dhellmann** [comment]
@Ashughorla This refactoring is going to change the implementation details of https://github.com/metal3-io/metal3-docs/pull/147 a little bit.

File: controllers/hardwareclassification_controller.go [hunk 1]
```diff
@@ -64,77 +63,140 @@ func (hcReconciler *HardwareClassificationReconciler) Reconcile(req ctrl.Request
...
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusUnMatched,
-			hwcc.Empty, hwcc.NOError)
-		hwcLog.Info("Updated profile status", "ProfileMatchStatus", hwcc.ProfileMatchStatusUnMatched)
+	if hardwareClassification.Status.ProfileMatchStatus != status {
+		hwcLog.Info("updating match status", "newValue", status)
+		hardwareClassification.Status.ProfileMatchStatus = status
+		err = hcReconciler.Status().Update(context.TODO(), hardwareClassification)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "failed to update status")
+		}
```
Comment (dhellmann) [inline controllers/hardwareclassification_controller.go:147]: With the new implementation there are no longer any errors to report, so we never set the status error fields. It wasn't really clear to me what we were considering an error before, so I just took out the code that was setting the error, but if there is something to set in that field we can replace it.

Comment (digambar15) [reply controllers/hardwareclassification_controller.go:147]: That's fine. Anyway, We are going to implement Error classification so we will cover all the error status related aspect as part of that feature.

**Review by digambar15 — APPROVED** [review]
Looks good

**Comment by digambar15** [comment]
Will wait for others to review one more day, I will merge the PR thereafter. 
Thanks @dhellmann. Nice implementation.

**Comment by digambar15** [comment]
@dhellmann  It seems some conflicts on this branch. Can you resolve it ?

**Comment by dhellmann** [comment]
I have rebased

**Comment by digambar15** [comment]
Thanks @dhellmann

**Pull request merged**